File: src/main/java/src/personas/Profesor.java
package personas;
import Calendario.*;
import java.util.ArrayList;


public class Profesor extends Persona {
	
	//atributos de instancia
	protected ArrayList<Materia> materias_Asignadas;
	
	//constructor
	public Profesor (String nombre, int ID, String Email) {
		super(nombre, ID, Email);
	}
	//metodos set y get
	
	
	public ArrayList<Materia> getMaterias_Asignadas(){
		return materias_Asignadas;
	}
	
	public void setMaterias_Asignadas(ArrayList<Materia> materiasAsignadas) {
		this.materias_Asignadas= materiasAsignadas;
	}
	
	//metodos de clase
	
	public void asignarMateria(Materia nuevaMateria) {
		materias_Asignadas.add(nuevaMateria);
	}
	public void retirarMateria(Materia Materia) {
		materias_Asignadas.remove(Materia);
	}
}

File: src python/gestorAplicacion.personas/Profesor.py
from gestorAplicacion.Calendario import *
from Persona import Persona
import numpy as np

class Profesor(Persona):
    def __init__(self, nombre, ID, Email):
        super().__init__(nombre, ID, Email)
        self.materias_Asignadas = []
        self.calificacionesDocente = []
        self.calificacionDocente = 0
        
    def setCalificacionDocente(self, calificacion):
        self.calificacionDocente = calificacion
        
    def getCalificacionDocente(self):
        return self.calificacionDocente
    
    def setCalificacionesDocente(self, calificacionDocente):
        self.calificacionesDocente = calificacionDocente
        
    def getCalificacionesDocente(self):
        return self.calificacionesDocente
    
    def getMaterias_Asignadas(self):
        return self.materias_Asignadas
    
    def setMaterias_Asignadas(self, materiasAsignadas):
        self.materias_Asignadas = materiasAsignadas
        
    def asignarMateria(self, nuevaMateria):
        self.materias_Asignadas.append(nuevaMateria)
        
    def retirarMateria(self, Materia):
        self.materias_Asignadas.remove(Materia)
        
    def ingresarCalificacion(self, calificacion):
        self.calificacionesDocente.append(calificacion)
        self.calificacionDocente = np.mean(self.calificacionesDocente)
    
    def ingresarCalificacion(self, calificacion):
        self.calificacionesDocente.append(calificacion)

    def retirarCalificacion(self, calificacion):    
        self.calificacionesDocente.remove(calificacion)

    def evaluacionDocente(self):
        totalCalificaciones = 0
        for calificacion in self.calificacionesDocente:
            totalCalificaciones += calificacion
        self.calificacionDocente = round((totalCalificaciones/len(self.calificacionesDocente))*10)/10.0

    def __str__(self):
        return "Docente " + self.getNombre()
